sql/src/main/java/io/crate/operation/collect/files/CSVLineParser.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to CRATE Technology GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.operation.collect.files;
+
+import com.fasterxml.jackson.databind.MappingIterator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.csv.CsvMapper;
+import com.fasterxml.jackson.dataformat.csv.CsvObjectReader;
+import com.fasterxml.jackson.dataformat.csv.CsvParser;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class CSVLineParser {
+
+    private static List<Object> keyList;
+    private static ObjectMapper objectMapper = new ObjectMapper();
+    private static CsvMapper csvMapper = new CsvMapper();
+    private static CsvObjectReader csvReader = csvMapper.enable(CsvParser.Feature.TRIM_SPACES)
```

Comment:
Are we certain that these can be shared across threads and have no state?
`static` attributes should also be marked as `final` if possible.

That the `keyList` is `static` doesn't look safe to me.

Reply:
`csvMapper` is never used apart from initializing the `csvReader`. So you could put it in a `static { }` block.

Even though `ObjectMapper` seems to be thread-safe, I'd also prefer if you could convert these to non-static fields.

Reply:
Ok 👍  I shall make these changes :)